crates/shared/src/order_quoting.rs

```diff
@@ -530,6 +548,78 @@ impl OrderQuoting for OrderQuoter {
     }
 }
 
+/// Handles whether or not quotes need to be verified.
+enum QuoteVerification {
+    /// Quotes get trusted blindly.
+    Never,
+    /// Enforce quotes only when the user has the required sell token balance.
+    /// If a user has the required sell token balance and a quote can still not
+    /// be verified the trade likely involves a weird token that solvers are not
+    /// able to handle correctly.
+    WhenFunded(Arc<dyn BalanceFetching>),
+}
+
+impl QuoteVerification {
+    /// Makes sure a quote was verified according to the configured rule.
+    async fn check(
+        &self,
+        estimate: &Estimate,
+        parameters: &QuoteParameters,
+        sell_amount: U256,
+    ) -> Result<(), CalculateQuoteError> {
+        let Some(verification) = &parameters.verification else {
```

Comment:
I find it a bit weird that the parameters can "request" verification and the quoter also has a "verification policy" (and they are somehow loosely enforced, ie both need to signal they want to verify for it to have an effect). Is there a way to unify those (and e.g. have `QuoteParameters.verification` become this enum instead of just an Option)

Reply:
Merged `--prefer-verified` and `--enforce-verified-quotes` into single argument `--quote-verfication` that takes an enum.
Also cleaned slightly cleaned up how we configure a trade simulator. `Web3` is now enabled when `--simulation-node-url` is set and if tenderly arguments are also set we turn it into a `Web3ThenTenderly` simulator.